watertap/property_models/coagulation_prop_pack.py

```diff
@@ -0,0 +1,640 @@
+                                   ('Liq', 'TDS'): 0.01*1000/31.4038218e-3}
+
+        self.seawater_mol_frac_dict = {('Liq', 'H2O'): 1,
+                                   ('Liq', 'TDS'): 0.01}
+
+        self.flow_mass_phase_comp = Var(
+            self.seawater_mass_frac_dict.keys(),
+            initialize=self.seawater_mass_flow_dict,
+            bounds=(1e-16, 100),
+            domain=NonNegativeReals,
+            units=pyunits.kg/pyunits.s,
+            doc='Mass flow rate')
+
+        self.temperature = Var(
+            initialize=298.15,
+            bounds=(273.15, 700),
+            domain=NonNegativeReals,
+            units=pyunits.degK,
+            doc='State temperature')
+
+        self.pressure = Var(
+            initialize=101325,
+            bounds=(0.9e5, 3e7),
+            domain=NonNegativeReals,
+            units=pyunits.Pa,
+            doc='State pressure')
+
+    # -----------------------------------------------------------------------------
+    # Property Methods
+    # these property methods build variables and constraints on demand (i.e. only when asked
+    # for by a user or unit model)
+    def _mass_frac_phase_comp(self):
+        self.mass_frac_phase_comp = Var(
+            self.seawater_mass_frac_dict.keys(),
+            initialize=self.seawater_mass_frac_dict,
+            bounds=(1e-16, 1.001),
+            units=pyunits.dimensionless,
+            doc='Mass fraction')
+
+        def rule_mass_frac_phase_comp(b, p, j):
+            if (p == 'Liq'):
+                return (b.mass_frac_phase_comp[p, j] == b.flow_mass_phase_comp[p, j] /
+                    (b.flow_mass_phase_comp['Liq', 'H2O'] +
+                        b.flow_mass_phase_comp['Liq', 'TDS'] +
+                            b.flow_mass_phase_comp['Liq', 'TSS']))
+            # Only 1 component in other phase, so mass frac is 1
+            else:
+                return (b.mass_frac_phase_comp[p, j] == 1.0)
+        self.eq_mass_frac_phase_comp = Constraint(self.seawater_mass_frac_dict.keys(), rule=rule_mass_frac_phase_comp)
+
+    def _dens_mass_phase(self):
+        self.dens_mass_phase = Var(
+            self.params.phase_list,
+            initialize={'Liq': 1000, 'Sol': 2600},
+            bounds=(5e2, 2e4),
+            units=pyunits.kg * pyunits.m**-3,
+            doc="Mass density")
+
+        def rule_dens_mass_phase(b, p):
+            if (p == 'Liq'):
+                return (b.dens_mass_phase[p] == (b.params.ref_dens_liq +
+                            b.params.dens_slope * b.mass_frac_phase_comp['Liq', 'TDS'] +
+                            b.params.dens_slope * b.mass_frac_phase_comp['Liq', 'TSS']) *
+                            (b.params.dens_param_A * b.temperature**2 + b.params.dens_param_B *
+                            b.temperature + b.params.dens_param_C) * (b.params.ref_pressure_correction +
+                            b.params.ref_pressure_slope * b.pressure) )
+            elif (p == 'Sol'):
```

Comment:
@aladshaw3  Can you add a quick comment on the source of this constraint? You may have it somewhere else already but I missed it.

Reply:
done